src/mesh/Router.cpp

```diff
@@ -355,13 +355,21 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
         if (moduleConfig.mqtt.enabled && isFromUs(p) && mqtt) {
             mqtt->onSend(*p, *p_decoded, chIndex);
         }
+#endif
+#if HAS_UDP_MULTICAST
+        // Only publish to UDP if we're the original transmitter of the packet
+        if (udpHandler && config.network.enabled_protocols & meshtastic_Config_NetworkConfig_ProtocolFlags_UDP_BROADCAST &&
+            isFromUs(p)) {
+            udpHandler->onSend(const_cast<meshtastic_MeshPacket *>(p), chIndex);
+        }
 #endif
         packetPool.release(p_decoded);
     }
-
 #if HAS_UDP_MULTICAST
-    if (udpHandler && config.network.enabled_protocols & meshtastic_Config_NetworkConfig_ProtocolFlags_UDP_BROADCAST) {
-        udpHandler->onSend(const_cast<meshtastic_MeshPacket *>(p));
+    // For packets that are already encrypted, we need to determine the channel from packet info
+    else if (udpHandler && config.network.enabled_protocols & meshtastic_Config_NetworkConfig_ProtocolFlags_UDP_BROADCAST &&
+             isFromUs(p)) {
+        udpHandler->onSend(const_cast<meshtastic_MeshPacket *>(p), p->channel);
```

Comment:
Here `p->channel` is not an index, but still the hash since it's encrypted, so looking up the channel in `onSend()` will fail.
To support PKI, we probably need to allow downlinking a hash of 0x00 if there is at least one channel with downlink enabled just like we do for MQTT.